examples/event_seg/simulated_data.py

```diff
@@ -0,0 +1,77 @@
+structure (with the same series of latent event patterns). An event
+segmentation is learned on the first dataset, and then we try to find the same
+series of events in other datasets. We measure how well we find the latent
+boundaries and the log-likelihood of the fits, and compare to a null model
+in which the event order is randomly shuffled.
+"""
+import brainiak.event_seg.event
+import numpy as np
+from scipy import stats
+import logging
+logging.basicConfig(level=logging.DEBUG)
+
+# Parameters for creating small simulated datasets
+V = 5
+K = 20
+T = 200
+T2 = 220
+
+# Generate the first dataset
+np.random.seed(1)
+event_means = np.random.rand(V, K)
+event_boundaries = np.random.randint(1, T, size=(K - 1))
+event_labels = np.zeros(T, dtype=int)
+event_labels[event_boundaries] = 1
+event_labels = np.cumsum(event_labels)
+
+simul_data = np.empty((V, T))
+for t in range(T):
+    simul_data[:, t] = stats.multivariate_normal.rvs(
+                       event_means[:, event_labels[t]], cov=0.01, size=1)
+
+simul_data = stats.zscore(simul_data, axis=1, ddof=1)
+
+# Find the events in this dataset
+simul_seg = brainiak.event_seg.event.EventSegment(K)
+segments = simul_seg.learn_events(simul_data)
+est_events = np.argmax(segments, axis=1)
+
+
+# Generate other datasets with the same underlying sequence of event
+# patterns, and try to find matching events
+test_loops = 10
+bound_match = np.empty((2, test_loops))
+LL = np.empty((2, test_loops))
+for test_i in range(test_loops):
+    # Generate data
+    event_boundaries2 = np.random.randint(1, T2, size=(K - 1))
+    event_labels2 = np.zeros(T2, dtype=int)
+    event_labels2[event_boundaries2] = 1
+    event_labels2 = np.cumsum(event_labels2)
+    simul_data2 = np.empty((V, T2))
+    for t in range(T2):
+        simul_data2[:, t] = stats.multivariate_normal.rvs(
+                            event_means[:, event_labels2[t]], cov=0.01, size=1)
+
+    # Find events matching previously-learned events
+    gamma, LL[0, test_i] = simul_seg.find_events(simul_data2)
+    est_events2 = np.argmax(gamma, axis=1)
+    bound_match[0, test_i] = 1 - np.sum(abs(np.diff(event_labels2) -
+                                            np.diff(est_events2))) / (2 * K)
+
+    # Run again, but with the order of events shuffled so that it no longer
+    # corresponds to the training data
+    gamma, LL[1, test_i] = simul_seg.find_events(simul_data2, scramble=True)
+    est_events2 = np.argmax(gamma, axis=1)
+    bound_match[1, test_i] = 1 - np.sum(abs(np.diff(event_labels2) -
+                                            np.diff(est_events2))) / (2 * K)
+
+# Across the testing datasets, print how well we identify the true event
+# boundaries and the log-likehoods in real vs. shuffled data
+print("Boundary match: {:.2} (null: {:.2})".format(
+    np.mean(bound_match[0, :]), np.mean(bound_match[1, :])))
+print("Log-likelihood: {:.3} (null: {:.3})".format(
+    np.mean(LL[0, :]), np.mean(LL[1, :])))
```

Comment:
Could you add a simple plot? We like colorful stuff. It's really a question, it's not mandatory.